src/main/java/frc/trigon/robot/poseestimation/apriltagcamera/AprilTagCamera.java

```diff
@@ -16,58 +15,43 @@ public class AprilTagCamera {
     private final Transform3d robotCenterToCamera;
-    private final double
-            thetaStandardDeviationExponent,
-            translationStandardDeviationExponent;
+    private final StandardDeviations standardDeviations;
     private final AprilTagCameraIO aprilTagCameraIO;
-    private double lastUpdatedTimestamp;
     private Pose2d robotPose = null;
 
     /**
      * Constructs a new AprilTagCamera.
      *
-     * @param aprilTagCameraType                   the type of camera
-     * @param name                                 the camera's name
-     * @param robotCenterToCamera                  the transform of the robot's origin point to the camera
-     * @param thetaStandardDeviationExponent       a calibrated gain for the standard theta deviations of the estimated robot pose
-     * @param translationStandardDeviationExponent a calibrated gain for the standard translation deviations of the estimated robot pose
+     * @param aprilTagCameraType  the type of camera
+     * @param name                the camera's name
+     * @param robotCenterToCamera the transform of the robot's origin point to the camera
+     * @param standardDeviations  the calibrated standard deviations for the camera's estimated pose
      */
     public AprilTagCamera(AprilTagCameraConstants.AprilTagCameraType aprilTagCameraType,
                           String name, Transform3d robotCenterToCamera,
-                          double thetaStandardDeviationExponent,
-                          double translationStandardDeviationExponent) {
+                          StandardDeviations standardDeviations) {
         this.name = name;
         this.robotCenterToCamera = robotCenterToCamera;
-        this.thetaStandardDeviationExponent = thetaStandardDeviationExponent;
-        this.translationStandardDeviationExponent = translationStandardDeviationExponent;
+        this.standardDeviations = standardDeviations;
 
-        if (RobotHardwareStats.isSimulation()) {
-            aprilTagCameraIO = AprilTagCameraConstants.AprilTagCameraType.SIMULATION_CAMERA.createIOFunction.apply(name);
+        aprilTagCameraIO = AprilTagCameraIO.generateIO(aprilTagCameraType, name);
+        if (RobotHardwareStats.isSimulation())
             aprilTagCameraIO.addSimulatedCameraToVisionSimulation(robotCenterToCamera);
-            return;
-        }
-        aprilTagCameraIO = aprilTagCameraType.createIOFunction.apply(name);
     }
 
     public void update() {
         aprilTagCameraIO.updateInputs(inputs);
 
-        robotPose = calculateBestRobotPose();
+        robotPose = getRobotPose();
         logCameraInfo();
     }
 
-    public boolean hasNewResult() {
-        return (inputs.hasResult && inputs.distanceFromBestTag != Double.POSITIVE_INFINITY) && isNewTimestamp();
-    }
-
-    public Pose2d getEstimatedRobotPose() {
+    public Pose2d getRobotPose() {
+        if (inputs.poseAmbiguity < AprilTagCameraConstants.MAXIMUM_AMBIGUITY)
```

Comment:
this check should be in `hasResult` (perhaps should be renamed to hasValidResult)